fetch.js

```diff
@@ -186,7 +186,9 @@
     }
 
     this.json = function() {
-      return this.text().then(JSON.parse)
+      return this.text().then(function(text) {
+        return JSON.parse(text)
+      })
```

Comment:
Is this a functional change, or just accommodates Firefox test behavior, or maybe both? It's disappointing to need a wrapper function just for tests.